### PR TITLE
Handle imread unreachable url

### LINKED ISSUE #3785: IO:  unhandled exception, in case the URL cannot be opened
## Description
image

## Way to reproduce
```python
def url_to_image(url):
    """download image from url and return it"""
    im_mat = io.imread(url)
    image = cv2.imdecode(im_mat, cv2.IMREAD_COLOR)
 
    # return the image
    return image
```

## Version information
```python
>>> from __future__ import print_function
>>> import sys; print(sys.version)
3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 23:09:28) [MSC v.1916 64 bit (AMD64)]
>>> import platform; print(platform.platform())
Windows-10-10.0.17134-SP0
>>> import skimage; print("scikit-image version: {}".format(skimage.__version__))
scikit-image version: 0.14.2
>>> import numpy; print("numpy version: {}".format(numpy.__version__))
```

```python
Traceback (most recent call last):
  File "C:\Python37\lib\site-packages\skimage\io\util.py", line 28, in file_or_url_context
    with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as f:
  File "C:\Python37\lib\tempfile.py", line 547, in NamedTemporaryFile
    (fd, name) = _mkstemp_inner(dir, prefix, suffix, flags, output_type)
  File "C:\Python37\lib\tempfile.py", line 258, in _mkstemp_inner
    fd = _os.open(file, flags, 0o600)
OSError: [Errno 22] Invalid argument: 'C:\\Users\\kelvin\\AppData\\Local\\Temp\\tmpegjt2y8r.png?mi_guid=CFCDF62B5115D31389F2893D3CCFFA5CB42BDE7F8CD696BB57B97B8BB567940F&mi_language=EN'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "imagecheck.py", line 28, in <module>
    print(url_to_image("http://www.movable-ink-6437.com/p/rp/0fd853751b1edbb0.png?mi_guid=CFCDF62B5115D31389F2893D3CCFFA5CB42BDE7F8CD696BB57B97B8BB567940F&mi_language=EN"))
  File "imagecheck.py", line 9, in url_to_image
    im_mat = io.imread(url)
  File "C:\Python37\lib\site-packages\skimage\io\_io.py", line 61, in imread
    with file_or_url_context(fname) as fname:
  File "C:\Python37\lib\contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "C:\Python37\lib\site-packages\skimage\io\util.py", line 34, in file_or_url_context
    os.remove(f.name)
UnboundLocalError: local variable 'f' referenced before assignment
```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Closes  #3785

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [X] Unit tests
- [X] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/io/util.py [hunk 1]
```diff
@@ -28,6 +28,8 @@ def file_or_url_context(resource_name):
                 f.write(u.read())
             # f must be closed before yielding
             yield f.name
+        except OSError:
+            raise ValueError('unreachable URL: {}'.format(resource_name))
```
Comment (hmaarrfk) [inline skimage/io/util.py]: what is the original error? could you reraise that one instead?

Comment (rfezzani) [reply skimage/io/util.py]: As mentioned in the original Issue (#3785), an `OSError: [Errno 22] Invalid argument` is raised followed by a `UnboundLocalError: local variable 'f' referenced before assignment`.
This error was not managed at all.

**Comment by hmaarrfk** [comment]
Could this be a windows only bug?

imageio:

<details>

```python
$ ipython                                                                                               
Python 3.7.3 | packaged by conda-forge | (default, Mar 27 2019, 23:01:00) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from imageio import imread                                                                      

In [2]: imread('http://fake_url.com/image.jpg')                                                         

---------------------------------------------------------------------------
ConnectionRefusedError                    Traceback (most recent call last)
~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1316                 h.request(req.get_method(), req.selector, req.data, headers,
-> 1317                           encode_chunked=req.has_header('Transfer-encoding'))
   1318             except OSError as err: # timeout error

~/miniconda3/envs/owl/lib/python3.7/http/client.py in request(self, method, url, body, headers, encode_chunked)
   1228         """Send a complete request to the server."""
-> 1229         self._send_request(method, url, body, headers, encode_chunked)
   1230 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in _send_request(self, method, url, body, headers, encode_chunked)
   1274             body = _encode(body, 'body')
-> 1275         self.endheaders(body, encode_chunked=encode_chunked)
   1276 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in endheaders(self, message_body, encode_chunked)
   1223             raise CannotSendHeader()
-> 1224         self._send_output(message_body, encode_chunked=encode_chunked)
   1225 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in _send_output(self, message_body, encode_chunked)
   1015         del self._buffer[:]
-> 1016         self.send(msg)
   1017 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in send(self, data)
    955             if self.auto_open:
--> 956                 self.connect()
    957             else:

~/miniconda3/envs/owl/lib/python3.7/http/client.py in connect(self)
    927         self.sock = self._create_connection(
--> 928             (self.host,self.port), self.timeout, self.source_address)
    929         self.sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)

~/miniconda3/envs/owl/lib/python3.7/socket.py in create_connection(address, timeout, source_address)
    726     if err is not None:
--> 727         raise err
    728     else:

~/miniconda3/envs/owl/lib/python3.7/socket.py in create_connection(address, timeout, source_address)
    715                 sock.bind(source_address)
--> 716             sock.connect(sa)
    717             # Break explicitly a reference cycle

ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

URLError                                  Traceback (most recent call last)
<ipython-input-2-403a4f1a57ea> in <module>
----> 1 imread('http://fake_url.com/image.jpg')

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/functions.py in imread(uri, format, **kwargs)
    219 
    220     # Get reader and read first
--> 221     reader = read(uri, format, "i", **kwargs)
    222     with reader:
    223         return reader.get_data(0)

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/functions.py in get_reader(uri, format, mode, **kwargs)
    134         format = formats[format]
    135     else:
--> 136         format = formats.search_read_format(request)
    137     if format is None:
    138         raise ValueError(

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/format.py in search_read_format(self, request)
    677         # Select the first that can
    678         for format in selected_formats:
--> 679             if format.can_read(request):
    680                 return format
    681 

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/format.py in can_read(self, request)
    183         Get whether this format can read data from the specified uri.
    184         """
--> 185         return self._can_read(request)
    186 
    187     def can_write(self, request):

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/plugins/pillow.py in _can_read(self, request)
    107                 factory, accept = Image.OPEN[self.plugin_id]
    108                 if accept:
--> 109                     if request.firstbytes and accept(request.firstbytes):
    110                         return True
    111 

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/request.py in firstbytes(self)
    455         """
    456         if self._firstbytes is None:
--> 457             self._read_first_bytes()
    458         return self._firstbytes
    459 

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/request.py in _read_first_bytes(self, N)
    464             # Prepare
    465             try:
--> 466                 f = self.get_file()
    467             except IOError:
    468                 if os.path.isdir(self.filename):  # A directory, e.g. for DICOM

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/request.py in get_file(self)
    366         elif self._uri_type in [URI_HTTP or URI_FTP]:
    367             assert not want_to_write  # This should have been tested in init
--> 368             self._file = urlopen(self.filename, timeout=5)
    369             make_file_object_support_noop_seeks(self._file)
    370 

~/miniconda3/envs/owl/lib/python3.7/site-packages/imageio/core/util.py in urlopen(*args, **kwargs)
     54         except ImportError:
     55             raise RuntimeError("Could not import urlopen.")
---> 56     return urlopen(*args, **kwargs)
     57 
     58 

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    220     else:
    221         opener = _opener
--> 222     return opener.open(url, data, timeout)
    223 
    224 def install_opener(opener):

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in open(self, fullurl, data, timeout)
    529         for processor in self.process_response.get(protocol, []):
    530             meth = getattr(processor, meth_name)
--> 531             response = meth(req, response)
    532 
    533         return response

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in http_response(self, request, response)
    639         if not (200 <= code < 300):
    640             response = self.parent.error(
--> 641                 'http', request, response, code, msg, hdrs)
    642 
    643         return response

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in error(self, proto, *args)
    561             http_err = 0
    562         args = (dict, proto, meth_name) + args
--> 563         result = self._call_chain(*args)
    564         if result:
    565             return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    501         for handler in handlers:
    502             func = getattr(handler, meth_name)
--> 503             result = func(*args)
    504             if result is not None:
    505                 return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in http_error_302(self, req, fp, code, msg, headers)
    753         fp.close()
    754 
--> 755         return self.parent.open(new, timeout=req.timeout)
    756 
    757     http_error_301 = http_error_303 = http_error_307 = http_error_302

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in open(self, fullurl, data, timeout)
    523             req = meth(req)
    524 
--> 525         response = self._open(req, data)
    526 
    527         # post-process response

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in _open(self, req, data)
    541         protocol = req.type
    542         result = self._call_chain(self.handle_open, protocol, protocol +
--> 543                                   '_open', req)
    544         if result:
    545             return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    501         for handler in handlers:
    502             func = getattr(handler, meth_name)
--> 503             result = func(*args)
    504             if result is not None:
    505                 return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in http_open(self, req)
   1343 
   1344     def http_open(self, req):
-> 1345         return self.do_open(http.client.HTTPConnection, req)
   1346 
   1347     http_request = AbstractHTTPHandler.do_request_

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1317                           encode_chunked=req.has_header('Transfer-encoding'))
   1318             except OSError as err: # timeout error
-> 1319                 raise URLError(err)
   1320             r = h.getresponse()
   1321         except:

URLError: <urlopen error [Errno 111] Connection refused>

In [3]: 
```
</details?
and with scikit-image 

<details>

```python
(owl) mark2@xps ~                                                                                       
$ ipython                                                                                               
Python 3.7.3 | packaged by conda-forge | (default, Mar 27 2019, 23:01:00) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from skimage.io import imread                                                                   

In [2]: imread('http://fake_url.com/image.jpg')                                                         
---------------------------------------------------------------------------
ConnectionRefusedError                    Traceback (most recent call last)
~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1316                 h.request(req.get_method(), req.selector, req.data, headers,
-> 1317                           encode_chunked=req.has_header('Transfer-encoding'))
   1318             except OSError as err: # timeout error

~/miniconda3/envs/owl/lib/python3.7/http/client.py in request(self, method, url, body, headers, encode_chunked)
   1228         """Send a complete request to the server."""
-> 1229         self._send_request(method, url, body, headers, encode_chunked)
   1230 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in _send_request(self, method, url, body, headers, encode_chunked)
   1274             body = _encode(body, 'body')
-> 1275         self.endheaders(body, encode_chunked=encode_chunked)
   1276 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in endheaders(self, message_body, encode_chunked)
   1223             raise CannotSendHeader()
-> 1224         self._send_output(message_body, encode_chunked=encode_chunked)
   1225 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in _send_output(self, message_body, encode_chunked)
   1015         del self._buffer[:]
-> 1016         self.send(msg)
   1017 

~/miniconda3/envs/owl/lib/python3.7/http/client.py in send(self, data)
    955             if self.auto_open:
--> 956                 self.connect()
    957             else:

~/miniconda3/envs/owl/lib/python3.7/http/client.py in connect(self)
    927         self.sock = self._create_connection(
--> 928             (self.host,self.port), self.timeout, self.source_address)
    929         self.sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)

~/miniconda3/envs/owl/lib/python3.7/socket.py in create_connection(address, timeout, source_address)
    726     if err is not None:
--> 727         raise err
    728     else:

~/miniconda3/envs/owl/lib/python3.7/socket.py in create_connection(address, timeout, source_address)
    715                 sock.bind(source_address)
--> 716             sock.connect(sa)
    717             # Break explicitly a reference cycle

ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

URLError                                  Traceback (most recent call last)
<ipython-input-2-403a4f1a57ea> in <module>
----> 1 imread('http://fake_url.com/image.jpg')

~/miniconda3/envs/owl/lib/python3.7/site-packages/skimage/io/_io.py in imread(fname, as_gray, plugin, flatten, **plugin_args)
     59             plugin = 'tifffile'
     60 
---> 61     with file_or_url_context(fname) as fname:
     62         img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
     63 

~/miniconda3/envs/owl/lib/python3.7/contextlib.py in __enter__(self)
    110         del self.args, self.kwds, self.func
    111         try:
--> 112             return next(self.gen)
    113         except StopIteration:
    114             raise RuntimeError("generator didn't yield") from None

~/miniconda3/envs/owl/lib/python3.7/site-packages/skimage/io/util.py in file_or_url_context(resource_name)
     27         try:
     28             with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as f:
---> 29                 u = urlopen(resource_name)
     30                 f.write(u.read())
     31             # f must be closed before yielding

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    220     else:
    221         opener = _opener
--> 222     return opener.open(url, data, timeout)
    223 
    224 def install_opener(opener):

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in open(self, fullurl, data, timeout)
    529         for processor in self.process_response.get(protocol, []):
    530             meth = getattr(processor, meth_name)
--> 531             response = meth(req, response)
    532 
    533         return response

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in http_response(self, request, response)
    639         if not (200 <= code < 300):
    640             response = self.parent.error(
--> 641                 'http', request, response, code, msg, hdrs)
    642 
    643         return response

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in error(self, proto, *args)
    561             http_err = 0
    562         args = (dict, proto, meth_name) + args
--> 563         result = self._call_chain(*args)
    564         if result:
    565             return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    501         for handler in handlers:
    502             func = getattr(handler, meth_name)
--> 503             result = func(*args)
    504             if result is not None:
    505                 return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in http_error_302(self, req, fp, code, msg, headers)
    753         fp.close()
    754 
--> 755         return self.parent.open(new, timeout=req.timeout)
    756 
    757     http_error_301 = http_error_303 = http_error_307 = http_error_302

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in open(self, fullurl, data, timeout)
    523             req = meth(req)
    524 
--> 525         response = self._open(req, data)
    526 
    527         # post-process response

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in _open(self, req, data)
    541         protocol = req.type
    542         result = self._call_chain(self.handle_open, protocol, protocol +
--> 543                                   '_open', req)
    544         if result:
    545             return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    501         for handler in handlers:
    502             func = getattr(handler, meth_name)
--> 503             result = func(*args)
    504             if result is not None:
    505                 return result

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in http_open(self, req)
   1343 
   1344     def http_open(self, req):
-> 1345         return self.do_open(http.client.HTTPConnection, req)
   1346 
   1347     http_request = AbstractHTTPHandler.do_request_

~/miniconda3/envs/owl/lib/python3.7/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1317                           encode_chunked=req.has_header('Transfer-encoding'))
   1318             except OSError as err: # timeout error
-> 1319                 raise URLError(err)
   1320             r = h.getresponse()
   1321         except:

URLError: <urlopen error [Errno 111] Connection refused>

In [3]: 
```

I think the error message is fine.

```
</details>

It seems to be even fine in skimage 0.14.2
```
In [3]: import skimage                                                                                  

In [4]: skimage.__version__                                                                             
Out[4]: '0.14.2'
```

**Comment by hmaarrfk** [comment]
Lets see the results for all different platforms here:
https://github.com/scikit-image/scikit-image/pull/3845

**Comment by rfezzani** [comment]
It in fact looks like a windows only bug: I can't reproduce it on linux. I observed:

- an `urllib.error.URLError` if the service is unknown
- an `urllib.error.HTTPError` in case of 403 or 404 HTTP errors.

I am surprised that the `test_imread_unreachable_url_handle` passed... Adding `URLError` and `HTTPError` to the errors handling can be a good option

**Comment by hmaarrfk** [comment]
I just can't reproduce the issue locally. For the URL used in the tests, the existing error message seems fine.
<details>
```python
Python 3.6.7 (default, Feb 28 2019, 07:28:18) [MSC v.1900 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from skimage.io import imread

In [2]: imread('http://fake_url.com/image.jpg')
---------------------------------------------------------------------------
ConnectionRefusedError                    Traceback (most recent call last)
~\Miniconda3\envs\mcam\lib\urllib\request.py in do_open(self, http_class, req, **http_conn_args)
   1317                 h.request(req.get_method(), req.selector, req.data, headers,
-> 1318                           encode_chunked=req.has_header('Transfer-encoding'))
   1319             except OSError as err: # timeout error

~\Miniconda3\envs\mcam\lib\http\client.py in request(self, method, url, body, headers, encode_chunked)
   1238         """Send a complete request to the server."""
-> 1239         self._send_request(method, url, body, headers, encode_chunked)
   1240

~\Miniconda3\envs\mcam\lib\http\client.py in _send_request(self, method, url, body, headers, encode_chunked)
   1284             body = _encode(body, 'body')
-> 1285         self.endheaders(body, encode_chunked=encode_chunked)
   1286

~\Miniconda3\envs\mcam\lib\http\client.py in endheaders(self, message_body, encode_chunked)
   1233             raise CannotSendHeader()
-> 1234         self._send_output(message_body, encode_chunked=encode_chunked)
   1235

~\Miniconda3\envs\mcam\lib\http\client.py in _send_output(self, message_body, encode_chunked)
   1025         del self._buffer[:]
-> 1026         self.send(msg)
   1027

~\Miniconda3\envs\mcam\lib\http\client.py in send(self, data)
    963             if self.auto_open:
--> 964                 self.connect()
    965             else:

~\Miniconda3\envs\mcam\lib\http\client.py in connect(self)
    935         self.sock = self._create_connection(
--> 936             (self.host,self.port), self.timeout, self.source_address)
    937         self.sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)

~\Miniconda3\envs\mcam\lib\socket.py in create_connection(address, timeout, source_address)
    723     if err is not None:
--> 724         raise err
    725     else:

~\Miniconda3\envs\mcam\lib\socket.py in create_connection(address, timeout, source_address)
    712                 sock.bind(source_address)
--> 713             sock.connect(sa)
    714             # Break explicitly a reference cycle

ConnectionRefusedError: [WinError 10061] No connection could be made because the target machine actively refused it

During handling of the above exception, another exception occurred:

URLError                                  Traceback (most recent call last)
<ipython-input-2-403a4f1a57ea> in <module>
----> 1 imread('http://fake_url.com/image.jpg')

~\Miniconda3\envs\mcam\lib\site-packages\skimage\io\_io.py in imread(fname, as_gray, plugin, flatten, **plugin_args)
     58             plugin = 'tifffile'
     59
---> 60     with file_or_url_context(fname) as fname:
     61         img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
     62

~\Miniconda3\envs\mcam\lib\contextlib.py in __enter__(self)
     79     def __enter__(self):
     80         try:
---> 81             return next(self.gen)
     82         except StopIteration:
     83             raise RuntimeError("generator didn't yield") from None

~\Miniconda3\envs\mcam\lib\site-packages\skimage\io\util.py in file_or_url_context(resource_name)
     25         try:
     26             with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as f:
---> 27                 u = urllib.request.urlopen(resource_name)
     28                 f.write(u.read())
     29             # f must be closed before yielding

~\Miniconda3\envs\mcam\lib\urllib\request.py in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    221     else:
    222         opener = _opener
--> 223     return opener.open(url, data, timeout)
    224
    225 def install_opener(opener):

~\Miniconda3\envs\mcam\lib\urllib\request.py in open(self, fullurl, data, timeout)
    530         for processor in self.process_response.get(protocol, []):
    531             meth = getattr(processor, meth_name)
--> 532             response = meth(req, response)
    533
    534         return response

~\Miniconda3\envs\mcam\lib\urllib\request.py in http_response(self, request, response)
    640         if not (200 <= code < 300):
    641             response = self.parent.error(
--> 642                 'http', request, response, code, msg, hdrs)
    643
    644         return response

~\Miniconda3\envs\mcam\lib\urllib\request.py in error(self, proto, *args)
    562             http_err = 0
    563         args = (dict, proto, meth_name) + args
--> 564         result = self._call_chain(*args)
    565         if result:
    566             return result

~\Miniconda3\envs\mcam\lib\urllib\request.py in _call_chain(self, chain, kind, meth_name, *args)
    502         for handler in handlers:
    503             func = getattr(handler, meth_name)
--> 504             result = func(*args)
    505             if result is not None:
    506                 return result

~\Miniconda3\envs\mcam\lib\urllib\request.py in http_error_302(self, req, fp, code, msg, headers)
    754         fp.close()
    755
--> 756         return self.parent.open(new, timeout=req.timeout)
    757
    758     http_error_301 = http_error_303 = http_error_307 = http_error_302

~\Miniconda3\envs\mcam\lib\urllib\request.py in open(self, fullurl, data, timeout)
    524             req = meth(req)
    525
--> 526         response = self._open(req, data)
    527
    528         # post-process response

~\Miniconda3\envs\mcam\lib\urllib\request.py in _open(self, req, data)
    542         protocol = req.type
    543         result = self._call_chain(self.handle_open, protocol, protocol +
--> 544                                   '_open', req)
    545         if result:
    546             return result

~\Miniconda3\envs\mcam\lib\urllib\request.py in _call_chain(self, chain, kind, meth_name, *args)
    502         for handler in handlers:
    503             func = getattr(handler, meth_name)
--> 504             result = func(*args)
    505             if result is not None:
    506                 return result

~\Miniconda3\envs\mcam\lib\urllib\request.py in http_open(self, req)
   1344
   1345     def http_open(self, req):
-> 1346         return self.do_open(http.client.HTTPConnection, req)
   1347
   1348     http_request = AbstractHTTPHandler.do_request_

~\Miniconda3\envs\mcam\lib\urllib\request.py in do_open(self, http_class, req, **http_conn_args)
   1318                           encode_chunked=req.has_header('Transfer-encoding'))
   1319             except OSError as err: # timeout error
-> 1320                 raise URLError(err)
   1321             r = h.getresponse()
   1322         except:

URLError: <urlopen error [WinError 10061] No connection could be made because the target machine actively refused it>

In [3]:
```

</details>

**Comment by rfezzani** [comment]
@hmaarrfk, THX for the review. As you suggested, I re-raised the existing error.

**Comment by hmaarrfk** [comment]
I mean ultimately, i can't recreate your issue.
https://github.com/scikit-image/scikit-image/pull/3845
See the results of your test there.
The stack trace already mentions something about a strange internet connection.

I think the original issue you referenced was fixed in 
https://github.com/scikit-image/scikit-image/pull/3642
and the fix should be in 0.15

**Comment by rfezzani** [comment]
All right, thank you. May be  #3785 should be closed then.

**Comment by hmaarrfk** [comment]
This really isn't to discourage your work.
The IO module does need some work to get all the edge cases. I just don't think you able, with the test you provided, to recreate the issue that you faced.

Did 0.15 solve your issue? Were you able to recreate your original issue with it?

Recreating these issues is legitimately difficult.

**Comment by rfezzani** [comment]
Don't worry, I'm not discouraged :-)
The issue is not mine, I just tried to fix the reported bug. If it is actually not a confirmed bug, no worry, it's even better ;-)

**Comment by hmaarrfk** [comment]
Yeah, I would totally encourage users to give us the full code requried to recreate their issue. If not, then we can't really be fixing things that we can't recreate.

Many issues are left open so as not to discourage the reporter to interact with scikit-image

**Comment by hmaarrfk** [comment]
@rfezzani can we close this PR?

**Comment by rfezzani** [comment]
Yes, no problem.
